lib/type1.g

```diff
@@ -374,42 +390,43 @@ end );
 ##
 BIND_GLOBAL( "Subtype2", function ( type, filter )
     local   new, i, save_flag;
-    save_flag := VAL_GVAR(_NEW_TYPE_READONLY);
-    ASS_GVAR(_NEW_TYPE_READONLY, false);
+#    save_flag := VAL_GVAR(_NEW_TYPE_READONLY);
+#    ASS_GVAR(_NEW_TYPE_READONLY, false);
     new := NEW_TYPE( TypeOfTypes,
                      type![1],
                      WITH_IMPS_FLAGS( AND_FLAGS(
                         type![2],
                         FLAGS_FILTER( filter ) ) ),
-                     type![ POS_DATA_TYPE ] );
-    for i in [ POS_FIRST_FREE_TYPE .. LEN_POSOBJ( type ) ] do
-        if IsBound( type![i] ) and not IsBound(new![i]) then
-            new![i] := type![i];
-        fi;
-    od;
-    MakeReadOnlyObj(new);
-    ASS_GVAR(_NEW_TYPE_READONLY, save_flag);
+                     type![ POS_DATA_TYPE ], type );
+    # for i in [ POS_FIRST_FREE_TYPE .. LEN_POSOBJ( type ) ] do
+    #     if IsBound( type![i] ) and not IsBound(new![i]) then
+    #         new![i] := type![i];
+    #     fi;
+    # od;
+    # MakeReadOnlyObj(new);
+    # ASS_GVAR(_NEW_TYPE_READONLY, save_flag);
     return new;
 end );
 
 
 BIND_GLOBAL( "Subtype3", function ( type, filter, data )
     local   new, i, save_flag;
```

Comment:
Note that `i` and `save_flag` are unused now. And `new` could also be removed, in favor of directly returning the result of `NEW_TYPE`. The same holds for Subtype3, SupType2, SupType3